server/src/main/java/org/elasticsearch/health/HealthIndicator.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.health;
+
+import org.elasticsearch.xcontent.ToXContentFragment;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+
+public record HealthIndicator(String name, String component, HealthStatus status, String summary, ToXContentFragment details)
+    implements
+        ToXContentObject {
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        builder.field("status", status);
+        builder.field("summary", summary);
+        builder.object("details", xContentBuilder -> details.toXContent(builder, params));
```

Comment:
Can we do this instead:

```suggestion
        builder.field("details", details, params);
```

and then make the details a non-fragment ToXContent?